# Income Statement analysis 

Horizontal analysis of the Given income statement

In [26]:
import pandas as pd
df=pd.read_excel(r'D:\Gradtwin\Gradtwin-Works\finance.xlsx')

In [27]:
df


Unnamed: 0  Revenue   COGS  Profit  Percentage calculation     TAX    Bonus  \
0      year1    51585  27697   23888                     NaN  4777.6  1194.40   
1      year2    53494  28429   25065                4.927160  5013.0  1253.25   
2      year3    55749  29200   26549                5.920606  5309.8  1327.45   
3      year4    50744  30859   19885              -25.100757  3977.0   994.25   

   Net Income  
0    20304.80  
1    21305.25  
2    22566.65  
3    16902.25

In [28]:
df.describe(include='all')

Unnamed: 0       Revenue          COGS        Profit  \
count           4      4.000000      4.000000      4.000000   
unique          4           NaN           NaN           NaN   
top         year1           NaN           NaN           NaN   
freq            1           NaN           NaN           NaN   
mean          NaN  52893.000000  29046.250000  23846.750000   
std           NaN   2224.634951   1355.381219   2856.772815   
min           NaN  50744.000000  27697.000000  19885.000000   
25%           NaN  51374.750000  28246.000000  22887.250000   
50%           NaN  52539.500000  28814.500000  24476.500000   
75%           NaN  54057.750000  29614.750000  25436.000000   
max           NaN  55749.000000  30859.000000  26549.000000   

        Percentage calculation          TAX        Bonus    Net Income  
count                 3.000000     4.000000     4.000000      4.000000  
unique                     NaN          NaN          NaN           NaN  
top                        NaN          NaN          NaN           NaN  
freq                       NaN          NaN          NaN           NaN  
mean                 -4.750997  4769.350000  1192.337500  20269.737500  
std                  17.630408   571.354563   142.838641   2428.256893  
min                 -25.100757  3977.000000   994.250000  16902.250000  
25%                 -10.086799  4577.450000  1144.362500  19454.162500  
50%                   4.927160  4895.300000  1223.825000  20805.025000  
75%                   5.423883  5087.200000  1271.800000  21620.600000  
max                   5.920606  5309.800000  1327.450000  22566.650000

In [29]:
df.sum(numeric_only=True)

Revenue                   211572.000000
COGS                      116185.000000
Profit                     95387.000000
Percentage calculation       -14.252991
TAX                        19077.400000
Bonus                       4769.350000
Net Income                 81078.950000
dtype: float64

In [31]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4 entries, 0 to 3
Data columns (total 8 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Unnamed: 0              4 non-null      object 
 1   Revenue                 4 non-null      int64  
 2   COGS                    4 non-null      int64  
 3   Profit                  4 non-null      int64  
 4   Percentage calculation  3 non-null      float64
 5   TAX                     4 non-null      float64
 6   Bonus                   4 non-null      float64
 7   Net Income              4 non-null      float64
dtypes: float64(4), int64(3), object(1)
memory usage: 388.0+ bytes


### Maximum profit is earned in year 3
### The average income earned through 4 years is  $20,269
### There is a average increases deviation of $2856 in each year
### Net Income earned through 4years is $81079
### there is percential increases is 2.78%

tasks.ezhil@gmail.com
